library/simulation/drone_sim.py

```diff
@@ -0,0 +1,61 @@
+import sys
+import numpy as np
+import cv2 as cv
+from nptyping import NDArray
+import struct
+
+from drone import Drone
+
+class DroneSim(Drone):
+    def __init__(self, racecar) -> None:
+        self.__racecar = racecar
+        self.__drone_image: NDArray[(480, 640, 3), np.uint8] = None
+        self.__is_drone_image_current: bool = False
+
+    def get_drone_image_no_copy(self) -> NDArray[(480, 640, 3), np.uint8]:
+        if not self.__is_drone_image_current:
+            self.__drone_image = self.__request_drone_image(False)
+            self.__is_drone_image_current = True
+
+        return self.__drone_image
+
+    def get_drone_image_async(self) -> NDArray[(480, 640, 3), np.uint8]:
+        return self.__request_drone_image(True)
+
+    def __update(self) -> None:

```

Comment:
Organize by public and private methods (ie move `__update` and `__request_drone_image` to the bottom)